modin/core/execution/ray/common/utils.py

```diff
@@ -186,6 +143,74 @@ def initialize_ray(
         NPartitions._put(num_cpus)
 
 
+def _object_store_memory() -> Optional[int]:
+    """
+    Get the object store memory we should start Ray with, in bytes.
+
+    If object store memory is set in the ``Memory`` Modin config variable:
+      return ``Memory``
+    else:
+      if on linux:
+        system_memory is available memory in /dev/shm
+      otherwise:
+        system_memory is psutil.virtual_memory().total
+      desired_memory = floor(_OBJECT_STORE_TO_SYSTEM_MEMORY_RATIO *
+                             system_memory / 1e9) * 1e9
+      if desired_memory == 0:
+        return None
+      else if (on Mac and desired_memory > the Ray mac object store size
+               limit L):
+        return L
+      else:
+        return desired_memory
+
+    Returns
+    -------
+    Optional[int]
+        The object store memory size in bytes, or None if we should use the Ray
+        default.
+    """
+    object_store_memory = Memory.get()
+    if object_store_memory is not None:
+        return object_store_memory
+    virtual_memory = psutil.virtual_memory().total
+    if sys.platform.startswith("linux"):
+        shm_fd = os.open("/dev/shm", os.O_RDONLY)
+        try:
+            shm_stats = os.fstatvfs(shm_fd)
+            system_memory = shm_stats.f_bsize * shm_stats.f_bavail
+            if system_memory / (virtual_memory / 2) < 0.99:
+                warnings.warn(
+                    f"The size of /dev/shm is too small ({system_memory} bytes). The required size "
+                    + f"at least half of RAM ({virtual_memory // 2} bytes). Please, delete files in /dev/shm or "
+                    + "increase size of /dev/shm with --shm-size in Docker. Also, you can set "
+                    + "the required memory size for each Ray worker in bytes to MODIN_MEMORY environment variable."
+                )
+        finally:
+            os.close(shm_fd)
+    else:
+        system_memory = virtual_memory
+    object_store_memory = int(
+        _OBJECT_STORE_TO_SYSTEM_MEMORY_RATIO * system_memory // 1e9 * 1e9
```

Comment:
is this equivalent to `int(_OBJECT_STORE_TO_SYSTEM_MEMORY_RATIO * system_memory)`?

Reply:
No, the `// 1e9` rounds down to the nearest gigabyte. I tried to change the calculation to reflect that, and also to clarify the calculation in the docstring.

Note that we need the `int` because floor-dividing a float by an int gives a float.

Reply:
sg. ty for the clarification!